src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -910,6 +933,90 @@ private void CreateFirstSegment(string segmentText)
             this.CreateDynamicPathSegment(null, identifier, parenthesisExpression);
         }
 
+        /// <summary>Creates the first <see cref="ODataPathSegment"/> for a request.</summary>
+        /// <param name="segmentText">The text of the segment.</param>
+        private bool CreateSegmentForEscapeFunction(string segmentText)
+        {
+            string identifier;
+            string parenthesisExpression;
+
+            ExtractSegmentIdentifierAndParenthesisExpression(segmentText, out identifier, out parenthesisExpression);
+
+            if (this.parsedSegments.Count == 0)
+            {
+                if (this.TryCreateSegmentForNavigationSource(identifier, parenthesisExpression))
+                {
+                    return true;
+                }
+
+                if (this.TryCreateSegmentForOperationImport(identifier, parenthesisExpression))
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                if (this.TryCreateFilterSegment(segmentText))
+                {
+                    return true;
+                }
+
+                ODataPathSegment previous = this.parsedSegments[this.parsedSegments.Count - 1];
+
+                // property if previous is single
+                if (previous.SingleResult)
+                {
+                    // if its not one of the recognized special segments, then it must be a property, type-segment, or key value.
+                    Debug.Assert(
+                        previous.TargetKind == RequestTargetKind.Resource
+                        || previous.TargetKind == RequestTargetKind.Dynamic,
+                        "previous.TargetKind(" + previous.TargetKind + ") can have properties");
+
+                    if (previous.TargetEdmType == null)
+                    {
+                        // A segment will correspond to a property in the object model;
+                        // if we are processing an open type, anything further in the
+                        // URI also represents an open type property.
+                        Debug.Assert(previous.TargetKind == RequestTargetKind.Dynamic, "For open properties, the target resource type must be null");
+                    }
+                    else
+                    {
+                        // if the segment corresponds to a declared property, handle it
+                        // otherwise, fall back to type-segments, actions, and dynamic/open properties
+                        IEdmProperty projectedProperty;
+                        if (this.TryBindProperty(identifier, out projectedProperty))
+                        {
+                            CheckSingleResult(previous.SingleResult, previous.Identifier);
+                            this.CreatePropertySegment(previous, projectedProperty, parenthesisExpression);
+
+                            return true;
+                        }
+                    }
+                }

```

Comment:
Maybe put this logic in a separate TryCreatePropertySegment to reduce duplicate code?